src/dodal/cli.py

```diff
@@ -62,16 +64,28 @@ def connect(beamline: str, all: bool, sim_backend: bool) -> None:
 
     print(f"Attempting connection to {beamline} (using {full_module_path})")
 
-    # Force all devices to be lazy (don't connect to PVs on instantiation) and do
-    # connection as an extra step, because the alternatives is handling the fact
-    # that only some devices may be lazy.
-    devices, instance_exceptions = make_all_devices(
-        full_module_path,
-        include_skipped=all,
-        fake_with_ophyd_sim=sim_backend,
-        wait_for_connection=False,
-    )
-    devices, connect_exceptions = _connect_devices(run_engine, devices, sim_backend)
+    mod = importlib.import_module(full_module_path)
+
+    # Don't connect devices as they're built and do connection as an extra step,
+    # because the alternatives is handling the fact that only some devices may
+    # be lazy.
+
+    if (manager := getattr(mod, device_manager, None)) and isinstance(
+        manager, DeviceManager
+    ):
+        devices, instance_exceptions, connect_exceptions = manager.build_and_connect(
+            mock=sim_backend,
+        )
```

Comment:
Don't we need a path provider on this path? Or will beamlines that need one have a pathprovider fixture that is overwritten by blueapi?

Reply:
That was the intention. I didn't want the CLI to assume anything about what was required by the beamline.